Add shared button styles + modify tag modal to match owner modal

ttannis opened the pull request:
In this PR I added 3 new specifications to the button configurations:
1. `btn-icon` for the "flat" buttons that render icons and are gray by default and a brand color on hover.
2. `btn-cancel` and `btn-save` which are currently used in our modal designs.

I've also update the modal in the `TagInput` to match the modal for `OwnerEditor`.

danwom commented on amundsen_application/static/css/_buttons-default.scss:
```diff
@@ -28,6 +28,37 @@
+  &.btn-icon {
+    border: none;
+    color: $gray-light;
+    -webkit-box-shadow: none !important;
```
I don't think we need to specify -webkit for box-shadow. 

https://caniuse.com/#search=box-shadow

danwom commented on amundsen_application/static/js/components/OwnerEditor/index.tsx:
```diff
@@ -145,7 +145,7 @@ class OwnerEditor extends React.Component<OwnerEditorProps, OwnerEditorState> {
             ref={ this.inputRef }
           />
-          <button className="btn btn-light add-btn" type="submit" aria-label="Add Item">
+          <button className="btn btn-light add-button" type="submit" aria-label="Add Item">
```
Shouldn't this also be a `btn-icon`?

danwom replied:
Oh nvm this is inside the modal

danwom commented on amundsen_application/static/css/_buttons-default.scss:
```diff
@@ -28,6 +28,37 @@
+    }
+  }
+
+  &.btn-icon {
```
I think the class name should indicate that this is a flat button with no border. We already have some buttons that have icons inside.

ttannis replied:
`btn-flat-icon`?

danwom commented on amundsen_application/static/js/components/OwnerEditor/styles.scss:
```diff
@@ -3,25 +3,10 @@
-  .btn.add-list-item {
-    color: $gray-light;
+  .btn.add-item-button {
     padding: 4px 1px;
```
I the padding and height should get baked into the `btn-icon` class, perhaps with a second modifier that indicates it comes with text.

ttannis replied:
So you're suggesting a new style, or modifying the existing one to apply the relevant styles to text if it exists?

ttannis commented:
@danwom 
1. Renamed new style to `btn-flat-icon`
2. Moved `padding` to `btn-flat-icon`, but kept the random `1px` left/right padding on the `add-item-button` because it just happens the line up a bit better because of how the icon is drawn.
3. Made the one-off changes to `AvatarLabel` for a 24px avatar and removing `width: 80%`
